src/main/java/com/ontotext/kafka/util/ValidateGraphDBConnection.java

```diff
@@ -0,0 +1,112 @@
+package com.ontotext.kafka.util;
+
+import com.ontotext.kafka.GraphDBSinkConfig;
+import org.apache.commons.io.IOUtils;
+import org.apache.kafka.common.config.Config;
+import org.apache.kafka.common.config.ConfigException;
+import org.apache.kafka.common.config.ConfigValue;
+import org.apache.kafka.common.config.types.Password;
+import org.eclipse.rdf4j.http.protocol.UnauthorizedException;
+import org.eclipse.rdf4j.repository.RepositoryConnection;
+import org.eclipse.rdf4j.repository.RepositoryException;
+import org.eclipse.rdf4j.repository.http.HTTPRepository;
+import org.json.JSONObject;
+
+import java.io.IOException;
+import java.net.URL;
+import java.nio.charset.Charset;
+import java.util.ArrayList;
+import java.util.stream.IntStream;
+
+import static com.ontotext.kafka.GraphDBSinkConfig.*;
+
+public class ValidateGraphDBConnection {
+
+	public static Config validateGraphDBConnection(Config validatedConnectorConfigs) {
+		ArrayList<ConfigValue> confValues = (ArrayList<ConfigValue>) validatedConnectorConfigs.configValues();
+
+		int serverIriId = getConfigIdByName(confValues, SERVER_IRI);
+		ConfigValue serverIri = confValues.get(serverIriId);
+
+		try {
+			validateGraphDBVersion(serverIri);
+		} catch (ConfigException e) {
+			serverIri.addErrorMessage(e.getMessage());
+			confValues.set(serverIriId, serverIri);
+			return validatedConnectorConfigs;
+		}
+
+		int repositoryId = getConfigIdByName(confValues, REPOSITORY);
+		ConfigValue repository = confValues.get(repositoryId);
+		int authTypeId = getConfigIdByName(confValues, AUTH_TYPE);
+		ConfigValue authType = confValues.get(authTypeId);
+
+		HTTPRepository testRepository = new HTTPRepository((String) serverIri.value(), (String) repository.value());
+		try {
+			validateGraphDBAuthAndRepo(confValues, testRepository, authType);
+		} catch (ConfigException e) {
+			authType.addErrorMessage(e.getMessage());
+			confValues.set(authTypeId, authType);
+		} catch (RepositoryException e){
+			repository.addErrorMessage(new ConfigException(REPOSITORY, repository.value(), e.getMessage() + ": Invalid repository").getMessage());
+			confValues.set(repositoryId, repository);
+		}
+
+		return validatedConnectorConfigs;
```

Comment:
why do you always return the config rather than simply throw the exception here

Reply:
because of the way the Kafka wants to handle configuration errors